### PR TITLE
Add presence

### PR DESCRIPTION
**Pull request by gieljnssns**
Can you please test if your Welcome camera still works.
I have custom components on my github page: https://github.com/gieljnssns/My_Home-Assistant/tree/master/homeassistant/custom_components

### CODE REVIEW

File: smart_home/__init__.py [hunk 1]
```diff
@@ -44,6 +45,12 @@ def todayStamps():
     today = int(time.mktime(time.strptime(today,"%Y-%m-%d")))
     return today, today+3600*24
 
+def apiCall():
```
Comment (jabesq) [inline smart_home/__init__.py]: This function is only for debugging purpose, I don't think it should be included in this PR

**Review by jabesq — CHANGES_REQUESTED** [review]

File: smart_home/Camera.py [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+        postParams = {
+            "access_token": self.getAuthToken
+            }
+        resp = postRequest(_GETHOMEDATA_REQ, postParams)
+        apiCall()
+        self.rawData = resp['body']
+        self.homes = {d['id']: d for d in self.rawData['homes']}
+        if not self.homes:
+            raise NoDevice("No camera available")
+        self.updateneeded = int(resp['time_server'] - 10)
```
Comment (jabesq) [inline smart_home/Camera.py]: This lib should not limit the API calls, it is the responsibility of the developer, who uses this lib, to be sure he does not reach one of the API limit (Per user limits: 50 requests every 10 seconds, 500 requests every hour). Developers might want to do a burst of API call every minute without reaching the API limit

**Comment by gieljnssns** [comment]
When I use lnetatmo 0.8.0 , I get this error:
```
16-12-06 08:54:50 ERROR (MainThread) [homeassistant.components.camera] Error while setting up platform netatmo
Traceback (most recent call last):
  File "/Users/giel/Github/home-assistant/homeassistant/helpers/entity_component.py", line 150, in _async_setup_platform
    entity_platform.add_entities, discovery_info
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/giel/Github/home-assistant/homeassistant/components/camera/netatmo.py", line 38, in setup_platform
    for camera_name in data.get_camera_names():
  File "/Users/giel/Github/home-assistant/homeassistant/components/netatmo.py", line 83, in get_camera_names
    self.update()
  File "/Users/giel/Github/home-assistant/homeassistant/util/__init__.py", line 296, in wrapper
    result = method(*args, **kwargs)
  File "/Users/giel/Github/home-assistant/homeassistant/components/netatmo.py", line 104, in update
    self.camera_data = lnetatmo.CameraData(self.auth, size=100)
  File "/Users/giel/.homeassistant/deps/smart_home/Camera.py", line 28, in __init__
    resp = postRequest(_GETHOMEDATA_REQ, postParams)
  File "/Users/giel/.homeassistant/deps/smart_home/__init__.py", line 33, in postRequest
    return json.loads(resp)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 65495 (char 65494)
16-12-06 08:54:50 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up platform netatmo
Traceback (most recent call last):
  File "/Users/giel/Github/home-assistant/homeassistant/helpers/entity_component.py", line 150, in _async_setup_platform
    entity_platform.add_entities, discovery_info
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/giel/Github/home-assistant/homeassistant/components/binary_sensor/netatmo.py", line 64, in setup_platform
    if data.get_camera_names() == []:
  File "/Users/giel/Github/home-assistant/homeassistant/components/netatmo.py", line 83, in get_camera_names
    self.update()
  File "/Users/giel/Github/home-assistant/homeassistant/util/__init__.py", line 296, in wrapper
    result = method(*args, **kwargs)
  File "/Users/giel/Github/home-assistant/homeassistant/components/netatmo.py", line 104, in update
    self.camera_data = lnetatmo.CameraData(self.auth, size=100)
  File "/Users/giel/.homeassistant/deps/smart_home/Camera.py", line 28, in __init__
    resp = postRequest(_GETHOMEDATA_REQ, postParams)
  File "/Users/giel/.homeassistant/deps/smart_home/__init__.py", line 33, in postRequest
    return json.loads(resp)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 65495 (char 65494)
```

**Comment by jabesq** [comment]
Can you try to replace 'resp = postRequest(_GETHOMEDATA_REQ, postParams)' by ' resp = postRequest(_GETHOMEDATA_REQ, postParams, body_size=None)' at line 25 of Welcome.py

Please also push your rebase so I can have the same env.

**Comment by gieljnssns** [comment]
my HA component changes
https://github.com/gieljnssns/home-assistant/tree/netatmo-presence

**Comment by jabesq** [comment]
Looks OK with body_size=None (the issue occurs if size of event was too large)

**Comment by gieljnssns** [comment]
Not ready to merge.

**Pull request closed without merging**